diff --git a/main/Makefile b/main/Makefile
index 718582df..b8fbdea7 100644
--- a/main/Makefile
+++ b/main/Makefile
@@ -8,13 +8,13 @@ MODULES =  module_wrf_top.F
 
 OBJS    =
 
-LIBPATHS = 
+LIBPATHS =
 
 include ../configure.wrf
 
 $(SOLVER)_wrf : wrf.o ../main/module_wrf_top.o
 	$(RANLIB) $(RLFLAGS) $(LIBWRFLIB)
-	$(LD) -o wrf.exe $(LDFLAGS) wrf.o ../main/module_wrf_top.o $(LIBWRFLIB) $(LIB) 
+	$(LD) -o wrf.exe $(LDFLAGS) wrf.o ../main/module_wrf_top.o $(LIBWRFLIB) $(LIB)
 
 $(SOLVER)_wrfplus : wrf.o ../main/module_wrf_top.o
 	$(RANLIB) $(RLFLAGS) $(LIBWRFLIB)
@@ -60,6 +60,9 @@ module_initialize : ../dyn_$(SOLVER)/module_initialize_$(IDEAL_CASE).o
 #	if [ -z "$(OMP)" ] ; then ( cd ../dyn_$(SOLVER) ;  $(MAKE) module_initialize_$(IDEAL_CASE).o ) ; fi
 # end of regtest changes
 
+# add dependencies so parallel build works
+real_$(SOLVER).o ndown_$(SOLVER).o nup_$(SOLVER).o tc_$(SOLVER).o ideal_nmm.o ideal_em.o ideal.o: ../dyn_$(SOLVER)/module_initialize_$(IDEAL_CASE).o
+
 clean:
 	@ echo 'use the clean script'
 
