diff --git a/Makefile b/Makefile
index cd1fec6b..343874e6 100644
--- a/Makefile
+++ b/Makefile
@@ -101,7 +101,7 @@ configcheck:
 	 echo "------------------------------------------------------------------------------" ; \
          exit 21 ; \
 	fi
- 
+
 
 framework_only : configcheck
 	$(MAKE) MODULE_DIRS="$(ALL_MODULES)" ext
@@ -147,7 +147,7 @@ wrfplus : configcheck
 	@echo "build started:   $(START_OF_COMPILE)"
 	@echo "build completed:" `date`
 
-all_wrfvar : 
+all_wrfvar :
 	$(MAKE) MODULE_DIRS="$(DA_WRFVAR_MODULES)" ext
 	$(MAKE) MODULE_DIRS="$(DA_WRFVAR_MODULES)" toolsdir
 	if [ $(CRTM) -ne 0 ] ; then \
@@ -494,7 +494,7 @@ em_scm_xy : wrf
 		ln -sf ../../run/LANDUSE.TBL . ; \
 		ln -sf ../../run/SOILPARM.TBL . ; \
 		ln -sf ../../run/VEGPARM.TBL . ; \
-		ln -sf ../../run/RRTM_DATA . ) 
+		ln -sf ../../run/RRTM_DATA . )
 	( cd run ; /bin/rm -f ideal.exe ; ln -s ../main/ideal.exe . )
 	( cd run ; if test -f namelist.input ; then \
 		/bin/cp -f namelist.input namelist.input.backup.`date +%Y-%m-%d_%H_%M_%S` ; fi ; \
@@ -525,11 +525,11 @@ convert_em : framework_only
             ( cd main ; $(MAKE) RLFLAGS="$(RLFLAGS)" MODULE_DIRS="$(ALL_MODULES)" convert_em ) ; \
         fi
 
-# Link wrf.exe and wrf_SST_ESMF.exe into 
-# test/em_esmf_exp when ESMF_COUPLING is set.  wrf.exe 
-# can be used for stand-alone testing in this case.  
-# wrf_SST_ESMF.exe is a coupled application.  Note that make 
-# target $(SOLVER)_wrf_SST_ESMF builds wrf_SST_ESMF.exe.  
+# Link wrf.exe and wrf_SST_ESMF.exe into
+# test/em_esmf_exp when ESMF_COUPLING is set.  wrf.exe
+# can be used for stand-alone testing in this case.
+# wrf_SST_ESMF.exe is a coupled application.  Note that make
+# target $(SOLVER)_wrf_SST_ESMF builds wrf_SST_ESMF.exe.
 em_real : wrf
 	@/bin/rm -f real.exe  > /dev/null 2>&1
 	@/bin/rm -f tc.exe    > /dev/null 2>&1
@@ -1003,7 +1003,7 @@ framework :
 	@ echo '--------------------------------------'
 	( cd frame ; $(MAKE) $(J) LLIST="$(LINKLIST)" framework ; \
           cd ../external/io_netcdf ; \
-          $(MAKE) NETCDFPATH="$(NETCDFPATH)" \
+          $(MAKE) NETCDFPATH="$(NETCDFPATH)" NETCDFFPATH="$(NETCDFFPATH)" \
                FC="$(FC) $(FCBASEOPTS) $(PROMOTION) $(FCDEBUG) $(OMP)" RANLIB="$(RANLIB)" \
                CPP="$(CPP)" LDFLAGS="$(LDFLAGS)" TRADFLAG="$(TRADFLAG)" ESMF_IO_LIB_EXT="$(ESMF_IO_LIB_EXT)" \
                LIB_LOCAL="$(LIB_LOCAL)" \
@@ -1133,7 +1133,7 @@ fseek_test :
 
 # rule used by configure to test if this will compile with netcdf4
 nc4_test:
-	@cd tools ; /bin/rm -f nc4_test.{exe,nc,o} ; $(SCC) -o nc4_test.exe nc4_test.c -I$(NETCDF)/include -L$(NETCDF)/lib $(USENETCDF) ; cd ..
+	@cd tools ; /bin/rm -f nc4_test.{exe,nc,o} ; $(SCC) -o nc4_test.exe nc4_test.c -I$(NETCDF)/include -L$(NETCDF)/lib64 $(USENETCDF) ; cd ..
 
 # rule used by configure to test if Fortran 2003 IEEE signaling is available
 fortran_2003_ieee_test:
@@ -1174,8 +1174,8 @@ toolsdir :
 
 #	( cd tools ; $(MAKE) CC_TOOLS="$(CC_TOOLS) -DIO_MASK_SIZE=$(IO_MASK_SIZE)" )
 
-# Use this target to build stand-alone tests of esmf_time_f90.  
-# Only touches external/esmf_time_f90/.  
+# Use this target to build stand-alone tests of esmf_time_f90.
+# Only touches external/esmf_time_f90/.
 esmf_time_f90_only :
 	@ echo '--------------------------------------'
 	( cd external/esmf_time_f90 ; $(MAKE) FC="$(FC) $(FCFLAGS)" CPP="$(CPP) -DTIME_F90_ONLY" tests )
diff --git a/arch/Config.pl b/arch/Config.pl
index ecccb670..cd5e3026 100644
--- a/arch/Config.pl
+++ b/arch/Config.pl
@@ -1,24 +1,25 @@
 #!/usr/bin/perl
 #
 # Configuration script for WRF prototype code
-# 
+#
 # Be sure to run as ./configure (to avoid getting a system configure command by mistake)
 #
 
 select((select(STDOUT), $|=1)[0]);
 $sw_perl_path = perl ;
 $sw_netcdf_path = "" ;
+$sw_netcdff_path = "" ;  # for netCDF 4.2 and greater, Fortran lib might be in a different path
 $sw_pnetcdf_path = "" ;
-$sw_hdf5_path=""; 
-$sw_phdf5_path=""; 
-$sw_jasperlib_path=""; 
-$sw_jasperinc_path=""; 
+$sw_hdf5_path="";
+$sw_phdf5_path="";
+$sw_jasperlib_path="";
+$sw_jasperinc_path="";
 $sw_esmflib_path="";
 $sw_esmfinc_path="";
 $sw_ctsm_mkfile_path="";
-$sw_ldflags=""; 
-$sw_compileflags=""; 
-$sw_opt_level=""; 
+$sw_ldflags="";
+$sw_compileflags="";
+$sw_opt_level="";
 $sw_rwordsize="\$\(NATIVE_RWORDSIZE\)";
 $sw_rttov_flag = "" ;
 $sw_rttov_inc = "" ;
@@ -40,7 +41,7 @@ $sw_dmparallel = "" ;
 $sw_ompparallel = "" ;
 $sw_stubmpi = "" ;
 $sw_usenetcdff = "" ;    # UNIDATA switches around library names a bit
-$sw_usenetcdf = "" ;    
+$sw_usenetcdf = "" ;
 $sw_time = "" ;          # name of a timer to time fortran compiles, e.g. timex or time
 $sw_ifort_r8 = 0 ;
 $sw_hdf5 = "-lhdf5_hl -lhdf5";
@@ -72,7 +73,7 @@ while ( substr( $ARGV[0], 0, 1 ) eq "-" )
   if ( substr( $ARGV[0], 1, 5 ) eq "gpfs=" )
   {
     $sw_gpfs_path = substr( $ARGV[0], 6 ) ;
-    if ( $sw_gpfs_path ne "" ) 
+    if ( $sw_gpfs_path ne "" )
       {
         if ( substr( $sw_gpfs_path, -1, 1 ) eq "/" )
           {
@@ -91,6 +92,10 @@ while ( substr( $ARGV[0], 0, 1 ) eq "-" )
           }
       }
   }
+  if ( substr( $ARGV[0], 1, 8 ) eq "netcdff=" )
+  {
+    $sw_netcdff_path = substr( $ARGV[0], 9 ) ;
+  }
   if ( substr( $ARGV[0], 1, 8 ) eq "pnetcdf=" )
   {
     $sw_pnetcdf_path = substr( $ARGV[0], 9 ) ;
@@ -118,6 +123,7 @@ while ( substr( $ARGV[0], 0, 1 ) eq "-" )
   if ( substr( $ARGV[0], 1, 11 ) eq "USENETCDFF=" )
   {
     $sw_usenetcdff = substr( $ARGV[0], 12 ) ;
+    $sw_usenetcdff =~ s/!/ /g ;
   }
   if ( substr( $ARGV[0], 1, 10 ) eq "USENETCDF=" )
   {
@@ -137,14 +143,14 @@ while ( substr( $ARGV[0], 0, 1 ) eq "-" )
   if ( substr( $ARGV[0], 1, 9 ) eq "wrf_core=" )
   {
     $sw_wrf_core = substr( $ARGV[0], 10 ) ;
-    if ( index ( $sw_wrf_core , "EM_CORE" ) > -1 ) 
+    if ( index ( $sw_wrf_core , "EM_CORE" ) > -1 )
     {
       $sw_em_core = "-DEM_CORE=1" ;
       $sw_da_core = "-DDA_CORE=0" ;
       $sw_wrfplus_core = "-DWRFPLUS=0" ;
       $sw_nmm_core = "-DNMM_CORE=0" ;
     }
-    if ( index ( $sw_wrf_core , "WRF_PLUS_CORE" ) > -1 ) 
+    if ( index ( $sw_wrf_core , "WRF_PLUS_CORE" ) > -1 )
     {
       $sw_em_core = "-DEM_CORE=1" ;
       $sw_da_core = "-DDA_CORE=0" ;
@@ -152,26 +158,26 @@ while ( substr( $ARGV[0], 0, 1 ) eq "-" )
       $sw_nmm_core = "-DNMM_CORE=0" ;
       $sw_dfi_radar = "-DDFI_RADAR=0" ;
     }
-    if ( index ( $sw_wrf_core , "DA_CORE" ) > -1 ) 
+    if ( index ( $sw_wrf_core , "DA_CORE" ) > -1 )
     {
       $sw_em_core = "-DEM_CORE=1" ;
       $sw_da_core = "-DDA_CORE=1" ;
       $sw_wrfplus_core = "-DWRFPLUS=0" ;
       $sw_nmm_core = "-DNMM_CORE=0" ;
     }
-    if ( index ( $sw_wrf_core , "4D_DA_CORE" ) > -1 ) 
+    if ( index ( $sw_wrf_core , "4D_DA_CORE" ) > -1 )
     {
       $sw_em_core = "-DEM_CORE=1" ;
       $sw_da_core = "-DDA_CORE=1" ;
       $sw_nmm_core = "-DNMM_CORE=0" ;
     }
-    if ( index ( $sw_wrf_core , "4D_DA_CORE" ) > -1 ) 
+    if ( index ( $sw_wrf_core , "4D_DA_CORE" ) > -1 )
     {
       $sw_em_core = "-DEM_CORE=1" ;
       $sw_da_core = "-DDA_CORE=1" ;
       $sw_nmm_core = "-DNMM_CORE=0" ;
     }
-    if ( index ( $sw_wrf_core , "NMM_CORE" ) > -1 ) 
+    if ( index ( $sw_wrf_core , "NMM_CORE" ) > -1 )
     {
       $sw_em_core = "-DEM_CORE=0" ;
       $sw_da_core = "-DDA_CORE=0" ;
@@ -185,14 +191,14 @@ while ( substr( $ARGV[0], 0, 1 ) eq "-" )
     $sw_compileflags =~ s/!/ /g ;
 #   look for each known option
     $where_index = index ( $sw_compileflags , "-DWRF_CHEM" ) ;
-    if ( $where_index eq -1 ) 
+    if ( $where_index eq -1 )
     {
       $WRFCHEM = 0 ;
     }
     else
     {
       $WRFCHEM = 1 ;
-    } 
+    }
   }
   if ( substr( $ARGV[0], 1, 11 ) eq "dmparallel=" )
   {
@@ -222,11 +228,11 @@ while ( substr( $ARGV[0], 0, 1 ) eq "-" )
  $sw_comms_lib = "" ;
  $sw_comms_include = "" ;
  $sw_dmparallelflag = "" ;
- $sw_nest_opt = "" ; 
+ $sw_nest_opt = "" ;
  $sw_comms_external = "gen_comms_serial module_dm_serial" ;
 
 
- if ( $sw_dmparallel eq "RSL_LITE" ) 
+ if ( $sw_dmparallel eq "RSL_LITE" )
  {
   $sw_fc = "\$(DM_FC)" ;
   $sw_cc = "\$(DM_CC)" ;
@@ -236,9 +242,9 @@ while ( substr( $ARGV[0], 0, 1 ) eq "-" )
   $sw_comms_include = "-I\$(WRF_SRC_ROOT_DIR)/external/RSL_LITE" ;
  }
 
-# The jasper library is required to build Grib2 I/O.  User must set 
-# environment variables JASPERLIB and JASPERINC to paths to library and 
-# include files to enable this feature prior to running configure.  
+# The jasper library is required to build Grib2 I/O.  User must set
+# environment variables JASPERLIB and JASPERINC to paths to library and
+# include files to enable this feature prior to running configure.
 
  $I_really_want_to_output_grib2_from_WRF = "FALSE" ;
 
@@ -247,8 +253,8 @@ while ( substr( $ARGV[0], 0, 1 ) eq "-" )
    printf "Configuring to use jasper library to build Grib2 I/O...\n" ;
    printf("  \$JASPERLIB = %s\n",$ENV{JASPERLIB});
    printf("  \$JASPERINC = %s\n",$ENV{JASPERINC});
-   $sw_jasperlib_path = $ENV{JASPERLIB}; 
-   $sw_jasperinc_path = $ENV{JASPERINC}; 
+   $sw_jasperlib_path = $ENV{JASPERLIB};
+   $sw_jasperinc_path = $ENV{JASPERINC};
    }
  else
    {
@@ -266,15 +272,15 @@ while ( substr( $ARGV[0], 0, 1 ) eq "-" )
 # When compiling DA and WRFPLUS code, we need to always use 8-byte reals.
  if ( $ENV{WRF_DA_CORE} eq "1" || $sw_da_core eq "-DDA_CORE=1" )
    {
-     $sw_rwordsize = "8";  
+     $sw_rwordsize = "8";
      if(defined $ENV{'CRTM'})
        {
        if ( $ENV{CRTM} ne "0" )
          {
          $sw_crtm_flag = "-DCRTM";
          }
-       } 
-     else 
+       }
+     else
        {
          {
          $sw_crtm_flag = "-DCRTM";
@@ -303,10 +309,10 @@ while ( substr( $ARGV[0], 0, 1 ) eq "-" )
 
  $sw_rwordsize = "8" if ( $sw_wrfplus_core eq "-DWRFPLUS=1" );
 
-# A separately-installed ESMF library is required to build the ESMF 
-# implementation of WRF IOAPI in external/io_esmf.  This is needed 
-# to couple WRF with other ESMF components.  User must set environment 
-# variables ESMFLIB and ESMFINC to paths ESMF to library and include 
+# A separately-installed ESMF library is required to build the ESMF
+# implementation of WRF IOAPI in external/io_esmf.  This is needed
+# to couple WRF with other ESMF components.  User must set environment
+# variables ESMFLIB and ESMFINC to paths ESMF to library and include
 # files to enable this feature prior to running configure.
  if ( $ENV{ESMFLIB} && $ENV{ESMFINC} )
    {
@@ -340,7 +346,7 @@ if ( $ENV{WRF_CTSM_MKFILE} ) {
 
 $validresponse = 0 ;
 
-if ( ($sw_wrf_core eq "WRF_PLUS_CORE") || ($sw_wrf_core eq "4D_DA_CORE") ) 
+if ( ($sw_wrf_core eq "WRF_PLUS_CORE") || ($sw_wrf_core eq "4D_DA_CORE") )
    { @platforms = qw ( serial dmpar ) ; }
    else
    { @platforms = qw ( serial smpar dmpar dm+sm ) ; }
@@ -352,15 +358,15 @@ until ( $validresponse ) {
 
   $opt = 1 ;
   $optstr = "";
-  open CONFIGURE_DEFAULTS, "< ./arch/configure.defaults" 
+  open CONFIGURE_DEFAULTS, "< ./arch/configure.defaults"
       or die "Cannot open ./arch/configure.defaults for reading" ;
   while ( <CONFIGURE_DEFAULTS> ) {
 
      $currline = $_;
      chomp $currline;
-     # Look for our platform in the configuration option header. 
+     # Look for our platform in the configuration option header.
      # If we're going to list it, print parallelism options
-     if ( substr( $currline, 0, 5 ) eq "#ARCH" && ( index( $currline, $sw_os ) >= 0 ) 
+     if ( substr( $currline, 0, 5 ) eq "#ARCH" && ( index( $currline, $sw_os ) >= 0 )
          && ( index( $currline, $sw_mach ) >= 0 ) ) {
         $optstr = substr($currline,6) ;
 
@@ -421,25 +427,25 @@ until ( $validresponse ) {
 
   if ( $response == -1 ) { exit ; }
 
-  if ( $response >= 1 && $response <= $opt ) 
+  if ( $response >= 1 && $response <= $opt )
   { $validresponse = 1 ; }
   else
   { printf("\nInvalid response (%d)\n",$response);}
-  $response_opt = $response ; 
+  $response_opt = $response ;
   chop $response_opt ;
 }
 printf "------------------------------------------------------------------------\n" ;
 
 $optchoice = $response ;
 if ( $response == 2 || $response == 3 ) {
-  if ( $ENV{'TERRAIN_AND_LANDUSE'} eq "1" && index($sw_wrf_core, "EM_CORE") > -1 ) { 
+  if ( $ENV{'TERRAIN_AND_LANDUSE'} eq "1" && index($sw_wrf_core, "EM_CORE") > -1 ) {
     $sw_terrain_and_landuse =" -DTERRAIN_AND_LANDUSE" ;
   } else {
     $sw_terrain_and_landuse =" -DLANDREAD_STUB=1" ;
   }
 } else {
   $sw_terrain_and_landuse =" -DLANDREAD_STUB=1" ;
-} 
+}
 open CONFIGURE_DEFAULTS, "cat ./arch/configure.defaults |"  ;
 $latchon = 0 ;
 while ( <CONFIGURE_DEFAULTS> )
@@ -458,6 +464,7 @@ while ( <CONFIGURE_DEFAULTS> )
   {
     $_ =~ s/CONFIGURE_PERL_PATH/$sw_perl_path/g ;
     $_ =~ s/CONFIGURE_NETCDF_PATH/$sw_netcdf_path/g ;
+    $_ =~ s/CONFIGURE_NETCDFF_PATH/$sw_netcdff_path/g ;
     $_ =~ s/CONFIGURE_PNETCDF_PATH/$sw_pnetcdf_path/g ;
     $_ =~ s/CONFIGURE_HDF5_PATH/$sw_hdf5_path/g ;
     $_ =~ s/CONFIGURE_PHDF5_PATH/$sw_phdf5_path/g ;
@@ -496,7 +503,7 @@ while ( <CONFIGURE_DEFAULTS> )
        $_ =~ s/#// ;
        $_ =~ s/#// ;
     }
-    if ( $sw_netcdf_path ) 
+    if ( $sw_netcdf_path )
       { $_ =~ s/CONFIGURE_WRFIO_NF/wrfio_nf/g ;
 	$_ =~ s:CONFIGURE_NETCDF_FLAG:-DNETCDF: ;
         if ( $ENV{NETCDF_LDFLAGS} ) {
@@ -504,16 +511,16 @@ while ( <CONFIGURE_DEFAULTS> )
         } elsif ( $sw_os eq "Interix" ) {
 	  $_ =~ s:CONFIGURE_NETCDF_LIB_PATH:\$\(WRF_SRC_ROOT_DIR\)/external/io_netcdf/libwrfio_nf.a -L$sw_netcdf_path/lib $sw_usenetcdff $sw_usenetcdf : ;
         } else {
-	  $_ =~ s:CONFIGURE_NETCDF_LIB_PATH:-L\$\(WRF_SRC_ROOT_DIR\)/external/io_netcdf -lwrfio_nf -L$sw_netcdf_path/lib $sw_usenetcdff $sw_usenetcdf : ;
+	  $_ =~ s:CONFIGURE_NETCDF_LIB_PATH:-L\$\(WRF_SRC_ROOT_DIR\)/external/io_netcdf -lwrfio_nf -L$sw_netcdff_path/lib -L$sw_netcdf_path/lib64 $sw_usenetcdff $sw_usenetcdf : ;
         }
 	 }
-    else                   
+    else
       { $_ =~ s/CONFIGURE_WRFIO_NF//g ;
 	$_ =~ s:CONFIGURE_NETCDF_FLAG::g ;
 	$_ =~ s:CONFIGURE_NETCDF_LIB_PATH::g ;
 	 }
 
-    if ( $sw_pnetcdf_path ) 
+    if ( $sw_pnetcdf_path )
       { $_ =~ s/CONFIGURE_WRFIO_PNF/wrfio_pnf/g ;
 	$_ =~ s:CONFIGURE_PNETCDF_FLAG:-DPNETCDF: ;
         if ( $sw_os eq "Interix" ) {
@@ -522,13 +529,13 @@ while ( <CONFIGURE_DEFAULTS> )
 	  $_ =~ s:CONFIGURE_PNETCDF_LIB_PATH:-L\$\(WRF_SRC_ROOT_DIR\)/external/io_pnetcdf -lwrfio_pnf -L$sw_pnetcdf_path/lib -lpnetcdf: ;
         }
 	 }
-    else                   
+    else
       { $_ =~ s/CONFIGURE_WRFIO_PNF//g ;
 	$_ =~ s:CONFIGURE_PNETCDF_FLAG::g ;
 	$_ =~ s:CONFIGURE_PNETCDF_LIB_PATH::g ;
 	 }
 
-    if ( $sw_hdf5_path ) 
+    if ( $sw_hdf5_path )
       { $_ =~ s:CONFIGURE_HDF5_LIB_PATH:-L$sw_hdf5_path/lib -lhdf5hl_fortran -lhdf5_hl -lhdf5_fortran -lhdf5 -lm -lz: ;
         $_ =~ s:CONFIGURE_HDF5_FLAG:-DHDF5: ;
          }
@@ -537,25 +544,25 @@ while ( <CONFIGURE_DEFAULTS> )
         $_ =~ s:CONFIGURE_HDF5_FLAG::g ;
          }
 
-    if ( $sw_phdf5_path ) 
+    if ( $sw_phdf5_path )
 
       { $_ =~ s/CONFIGURE_WRFIO_PHDF5/wrfio_phdf5/g ;
 	$_ =~ s:CONFIGURE_PHDF5_FLAG:-DPHDF5: ;
 	$_ =~ s:CONFIGURE_PHDF5_LIB_PATH:-L\$\(WRF_SRC_ROOT_DIR\)/external/io_phdf5 -lwrfio_phdf5 -L$sw_phdf5_path/lib -lhdf5_fortran -lhdf5 -lm -lz -L$sw_phdf5_path/lib -lsz: ;
 	 }
-    else                   
+    else
       { $_ =~ s/CONFIGURE_WRFIO_PHDF5//g ;
 	$_ =~ s:CONFIGURE_PHDF5_FLAG::g ;
 	$_ =~ s:CONFIGURE_PHDF5_LIB_PATH::g ;
 	 }
 
-    if ( $sw_jasperlib_path && $sw_jasperinc_path ) 
+    if ( $sw_jasperlib_path && $sw_jasperinc_path )
       { $_ =~ s/CONFIGURE_WRFIO_GRIB2/wrfio_grib2/g ;
         $_ =~ s:CONFIGURE_GRIB2_FLAG:-DGRIB2:g ;
         $_ =~ s:CONFIGURE_GRIB2_INC:-I$sw_jasperinc_path:g ;
         $_ =~ s:CONFIGURE_GRIB2_LIB:-L\$\(WRF_SRC_ROOT_DIR\)/external/io_grib2 -lio_grib2 -L$sw_jasperlib_path -ljasper:g ;
       }
-    else                   
+    else
       { $_ =~ s/CONFIGURE_WRFIO_GRIB2//g ;
         $_ =~ s:CONFIGURE_GRIB2_FLAG::g ;
         $_ =~ s:CONFIGURE_GRIB2_INC::g ;
@@ -563,7 +570,7 @@ while ( <CONFIGURE_DEFAULTS> )
       }
 
    if ( $sw_terrain_and_landuse )
-     { 
+     {
         $_ =~ s/CONFIGURE_TERRAIN_AND_LANDUSE/$sw_terrain_and_landuse/g;
      }
    else
@@ -582,7 +589,7 @@ while ( <CONFIGURE_DEFAULTS> )
       $_ =~ s:ESMFIOLIB:\$\(ESMF_F90LINKPATHS\) \$\(ESMF_F90ESMFLINKLIBS\) -L\$\(WRF_SRC_ROOT_DIR\)/external/io_esmf -lwrfio_esmf: ;
       $_ =~ s:ESMFIOEXTLIB:\$\(ESMF_IO_LIB\): ;
 
-     
+
       $_ =~ s:ESMFLIBFLAG:\$\(ESMF_LDFLAG\):g ;
 #      $_ =~ s:ESMFINCLUDEGOESHERE:'include $(ESMFLIB)/esmf.mk': ;
 
@@ -627,11 +634,11 @@ while ( <CONFIGURE_DEFAULTS> )
      if ( $ENV{NETCDF4} )
        { if ( $ENV{NETCDF4} eq "1" )
            {
-             if ( /(^ARCH_LOCAL.*=|^TRADFLAG.*=)/ ) 
-               { $_  =~ s/\r|\n//g; 
-                 $_ .= " \$\(NETCDF4_IO_OPTS\)\n" ; 
+             if ( /(^ARCH_LOCAL.*=|^TRADFLAG.*=)/ )
+               { $_  =~ s/\r|\n//g;
+                 $_ .= " \$\(NETCDF4_IO_OPTS\)\n" ;
                }
-             if (/^LIB.*=/) 
+             if (/^LIB.*=/)
                { $_  =~ s/(\\?)[\r\n]/ \$\(NETCDF4_DEP_LIB\)\1\n/g ;
                }
            }
@@ -644,16 +651,16 @@ while ( <CONFIGURE_DEFAULTS> )
     }
   }
 
-# nesting support 
+# nesting support
 # 0 = no nesting (only selectable for serial and smpar)
 # 1 = basic nesting (serial and smpar compile with RSL_LITE and STUBMPI; dmpar and dm+sm use RSL_LITE and MPI)
 # 2 = nesting with prescribed moves  (add -DMOVE_NESTS to ARCHFLAGS)
-# 3 = nesting with prescribed moves  (add -DMOVE_NESTS and -DVORTEX_CENTER to ARCHFLAGS) 
+# 3 = nesting with prescribed moves  (add -DMOVE_NESTS and -DVORTEX_CENTER to ARCHFLAGS)
 
   for $paropt ( @platforms )
   {
-    if ( substr( $_, 0, 5 ) eq "#ARCH" && $latchon == 0 
-          && ( index( $_, $sw_os ) >= 0 ) && ( index( $_, $sw_mach ) >= 0 ) 
+    if ( substr( $_, 0, 5 ) eq "#ARCH" && $latchon == 0
+          && ( index( $_, $sw_os ) >= 0 ) && ( index( $_, $sw_mach ) >= 0 )
           && ( index($_, $paropt) >= 0 ) )
     {
       # We are cycling through the configure.defaults file again.
@@ -693,7 +700,7 @@ while ( <CONFIGURE_DEFAULTS> )
                printf "Compile for nesting? (1=basic, 2=preset moves, 3=vortex following) [default 1]: " ;
              }
              $response = <STDIN> ;
-          } 
+          }
           printf "\n" ;
           lc $response ;
           chop $response ;
@@ -702,13 +709,13 @@ while ( <CONFIGURE_DEFAULTS> )
           else
             { printf("\nInvalid response (%d)\n",$response);}
         }
-        if ( $response == "" ) { 
+        if ( $response == "" ) {
           if ( ( $paropt eq 'serial' || $paropt eq 'smpar' ) ) { $response = 0 ; }
           else                                                 { $response = 1 ; }
         }
         if ( $response == 0 ) {
           if ( ! ( $paropt eq 'serial' || $paropt eq 'smpar' ) ) { $response = 1 ; }
-        } 
+        }
         $response_nesting = $response ;
         if ( ( $response == 1 ) || ( $response == 2 ) || ( $response == 3 ) ) {
           if ( ( $paropt eq 'serial' || $paropt eq 'smpar' ) ) {   # nesting without MPI
@@ -722,22 +729,22 @@ while ( <CONFIGURE_DEFAULTS> )
             $sw_dmparallel = "RSL_LITE" ;
             $sw_dmparallelflag = "-DDM_PARALLEL" ;
           }
-        } 
+        }
         if ( $response == 2 ) {
-          $sw_nest_opt = "-DMOVE_NESTS" ; 
+          $sw_nest_opt = "-DMOVE_NESTS" ;
           if ( $ENV{'TERRAIN_AND_LANDUSE'} eq "1" ) {
             $sw_terrain_and_landuse =" -DTERRAIN_AND_LANDUSE" ;
-            $sw_nest_opt = $sw_nest_opt . $sw_terrain_and_landuse; 
-          }  
+            $sw_nest_opt = $sw_nest_opt . $sw_terrain_and_landuse;
+          }
         } elsif ( $response == 3 ) {
-          $sw_nest_opt = "-DMOVE_NESTS -DVORTEX_CENTER" ; 
+          $sw_nest_opt = "-DMOVE_NESTS -DVORTEX_CENTER" ;
           if ( $ENV{'TERRAIN_AND_LANDUSE'} eq "1" ) {
             $sw_terrain_and_landuse =" -DTERRAIN_AND_LANDUSE" ;
-            $sw_nest_opt = $sw_nest_opt . $sw_terrain_and_landuse; 
+            $sw_nest_opt = $sw_nest_opt . $sw_terrain_and_landuse;
           }
         }
         if ( $paropt eq 'smpar' || $paropt eq 'dm+sm' ) { $sw_ompparallel = "OMP" ; }
-        if ( $paropt eq 'dmpar' || $paropt eq 'dm+sm' ) { 
+        if ( $paropt eq 'dmpar' || $paropt eq 'dm+sm' ) {
           if ( $sw_os ne "CYGWIN_NT" ) {
             $sw_comms_lib = "\$(WRF_SRC_ROOT_DIR)/external/RSL_LITE/librsl_lite.a" ;
             if ( $sw_wrf_core eq "4D_DA_CORE" )
@@ -914,7 +921,7 @@ while ( <ARCH_PREAMBLE> )
   if ( $sw_curl_path ne "" )
     { if (/^CURL.*=/)
         { $_  =~ s/\r|\n//g;
-          if ( $sw_curl_path ne "DEFAULT" ) 
+          if ( $sw_curl_path ne "DEFAULT" )
             { $_ .= " -L" . $sw_curl_path ; }
           $_ .= " " . $sw_curl_lib . "\n" ;
         }
@@ -949,5 +956,3 @@ close CONFIGURE_WRF ;
 
 printf "Configuration successful! \n" ;
 printf "------------------------------------------------------------------------\n" ;
-
-
diff --git a/arch/conf_tokens b/arch/conf_tokens
index 2695b308..7b1f7df2 100644
--- a/arch/conf_tokens
+++ b/arch/conf_tokens
@@ -5,6 +5,7 @@
        CONFIGURE_DMPARALLEL
 CONFIGURE_RWORDSIZE
 CONFIGURE_NETCDF_FLAG
+CONFIGURE_NETCDFF_FLAG
 CONFIGURE_GRIB2_FLAG
 CONFIGURE_NETCDF_LIB_PATH
 CONFIGURE_GRIB2_LIB
@@ -13,4 +14,5 @@ CONFIGURE_COMPILEFLAGS
 CONFIGURE_WRFIO_NF 
 CONFIGURE_WRFIO_GRIB2 
 CONFIGURE_NETCDF_PATH
+CONFIGURE_NETCDFF_PATH
 CONFIGURE_GRIB2_INC
diff --git a/arch/configure.defaults b/arch/configure.defaults
index e9f26f0a..16b713cb 100644
--- a/arch/configure.defaults
+++ b/arch/configure.defaults
@@ -769,8 +769,8 @@ SCC             =       gcc
 CCOMP           =       gcc
 DM_FC           =       mpif90 -f90=$(SFC)
 DM_CC           =       mpicc -cc=$(SCC)
-FC              =       CONFIGURE_FC
-CC              =       CONFIGURE_CC
+FC              =       ftn
+CC              =       cc
 LD              =       $(FC)
 RWORDSIZE       =       CONFIGURE_RWORDSIZE
 PROMOTION       =       #-fdefault-real-8
@@ -1602,6 +1602,7 @@ CC_TOOLS        =       $(SCC)
 
 LIB_EXTERNAL    = \
                      ../external/io_netcdf/libwrfio_nf.a CONFIGURE_NETCDF_PATH/lib/libnetcdf.lib \
+		     CONFIGURE_NETCDFF_PATH/lib/libnetcdff.lib \
                      ../external/wavelet/libWavelet.a ../external/wavelet/lib_wavelet.a
 ESMF_IO_LIB     =    ../external/esmf_time_f90/libesmf_time.a
 LIB_BUNDLED     = \
diff --git a/arch/postamble b/arch/postamble
index 152010fe..89fdea3b 100644
--- a/arch/postamble
+++ b/arch/postamble
@@ -52,7 +52,7 @@ INCLUDE_MODULES =    $(MODULE_SRCH_FLAG) \
                       -I$(WRF_SRC_ROOT_DIR)/phys \
                       -I$(WRF_SRC_ROOT_DIR)/wrftladj \
                       -I$(WRF_SRC_ROOT_DIR)/chem -I$(WRF_SRC_ROOT_DIR)/inc \
-                      -I$(NETCDFPATH)/include \
+                      -I$(NETCDFPATH)/include -I$(NETCDFFPATH)/include \
                       CONFIGURE_RTTOV_INC CONFIGURE_CTSM_INC
 REGISTRY        =    Registry
 CC_TOOLS_CFLAGS = CONFIGURE_NMM_CORE
@@ -62,6 +62,7 @@ LDFLAGS         =    $(OMP) $(FCFLAGS) $(LDFLAGS_LOCAL) CONFIGURE_LDFLAGS
 ENVCOMPDEFS     =    CONFIGURE_COMPILEFLAGS
 CPPFLAGS        =    $(ARCHFLAGS) $(ENVCOMPDEFS) -I$(LIBINCLUDE) $(TRADFLAG) CONFIGURE_COMMS_INCLUDE
 NETCDFPATH      =    CONFIGURE_NETCDF_PATH
+NETCDFFPATH     =    CONFIGURE_NETCDFF_PATH
 HDF5PATH        =    CONFIGURE_HDF5_PATH
 WRFPLUSPATH     =    CONFIGURE_WRFPLUS_PATH
 RTTOVPATH       =    CONFIGURE_RTTOV_PATH
@@ -93,13 +94,13 @@ module_dm_rsllite :
 
 wrfio_nf : 
 	( cd $(WRF_SRC_ROOT_DIR)/external/io_netcdf ; \
-          make $(J) NETCDFPATH="$(NETCDFPATH)" RANLIB="$(RANLIB)" CPP="$(CPP)" \
+	  make $(J) NETCDFPATH="$(NETCDFPATH)" NETCDFFPATH="$(NETCDFFPATH)" RANLIB="$(RANLIB)" CPP="$(CPP)" \
           CC="$(SCC)" CFLAGS="$(CFLAGS)" \
           FC="$(SFC) $(PROMOTION) $(OMP) $(FCFLAGS)" TRADFLAG="$(TRADFLAG)" AR="$(AR)" ARFLAGS="$(ARFLAGS)" )
 
 wrfio_pnf : 
 	( cd $(WRF_SRC_ROOT_DIR)/external/io_pnetcdf ; \
-          make $(J) NETCDFPATH="$(PNETCDFPATH)" RANLIB="$(RANLIB)" CPP="$(CPP) $(ARCHFLAGS)" \
+	  make $(J) NETCDFPATH="$(PNETCDFPATH)" NETCDFFPATH="$(PNETCDFPATH)" RANLIB="$(RANLIB)" CPP="$(CPP) $(ARCHFLAGS)" \
           FC="$(FC) $(PROMOTION) $(OMP) $(FCFLAGS)" TRADFLAG="$(TRADFLAG)" AR="$(AR)" ARFLAGS="$(ARFLAGS)" )
 
 wrfio_grib_share :
diff --git a/configure b/configure
index ed9f10c2..6de8b5b0 100755
--- a/configure
+++ b/configure
@@ -50,7 +50,7 @@ if [ -n "$print_usage" ] ; then
   echo '-help print this message'
   echo '*****************************************************************************'
   echo ' '
-  exit 
+  exit
 fi
 
 if `pwd | grep ' ' > /dev/null ` ; then
@@ -123,7 +123,7 @@ do
     else
       echo "$ac_t""found $ac_prog_where ($PERL)" 1>&1
     fi
-    PERLFULLPATH=$ac_prog_where  
+    PERLFULLPATH=$ac_prog_where
   else
     if test -z "$ac_echo_test" -a 1 = 1 ; then
       echo "$ac_t""no"
@@ -135,16 +135,16 @@ do
 done
 
 if test -z "$PERL" ; then
-    # We have to set this outside of the loop lest the first failure in 
+    # We have to set this outside of the loop lest the first failure in
     # PROGRAM_CHECK set the value (which then terminates the effect of the
-    # loop, since autoconf macros only set values that are null, they 
+    # loop, since autoconf macros only set values that are null, they
     # don't override them
     PERL=""
 fi
 
 if test -n "$PERL" ; then
     PERL="$PERL"
-    
+
     perlversion=`$PERL -v | grep 'This is perl' | \
        sed -e 's/^.*v[a-z ]*\([0-9]\).*$/\1/'`
 
@@ -156,7 +156,7 @@ if test -n "$PERL" ; then
         echo "Configure believes that $PERL is version $perlversion ."
         PERL=""
     fi
-    
+
 fi
 
 if  test -z "$NETCDF"  ; then
@@ -183,10 +183,14 @@ if [ -n "$NETCDF" ] ; then
   if [ -f "$NETCDF/lib/libnetcdff.a" -o -f "$NETCDF/lib/libnetcdff.so" -o -f "$NETCDF/lib/libnetcdff.dll.a" ] ; then
     USENETCDFF="-lnetcdff"
   else
-    USENETCDFF=" "
+    if [ -f "$NETCDFF/lib/libnetcdff.a" -o -f "$NETCDFF/lib/libnetcdff.so" -o -f "$NETCDFF/lib64/libnetcdff.so" -o -f "$NETCDFF/lib64/libnetcdff.so" ] ; then
+      USENETCDFF="-L$NETCDFF/lib!-L$NETCDFF/lib64!-lnetcdff" # ! will be replaced with space
+    else
+      USENETCDFF=" "
+    fi
   fi
-  if [ -f "$NETCDF/lib/libnetcdf.a" -o -f "$NETCDF/lib/libnetcdf.so" -o -f "$NETCDF/lib/libnetcdf.dll.a" ] ; then
-    USENETCDF="-lnetcdf"
+  if [ -f "$NETCDF/lib/libnetcdf.a" -o -f "$NETCDF/lib/libnetcdf.so" -o -f "$NETCDF/lib64/libnetcdf.a" -o -f "$NETCDF/lib64/libnetcdf.so" ] ; then
+      USENETCDF="-lnetcdf"
   else
     USENETCDF=" "
   fi
@@ -258,7 +262,7 @@ if [ -n "$NETCDF4" ] ; then
     fi
     for P in "$HDF5_PATH" "$ZLIB_PATH" "$GPFS_PATH" "$CURL_PATH"
     do
-      if [ "$P" != '' -a "$P" != "DEFAULT" ] ; then 
+      if [ "$P" != '' -a "$P" != "DEFAULT" ] ; then
         if [ "${P#${P%?}}" = "/" ] ; then
           P=`echo $P | sed 's/\/$//'`
         fi
@@ -286,7 +290,7 @@ if [ -n "$NETCDF4" ] ; then
         NETCDF=${NETCDF%/*}/$lastword
       fi
       buff="`ls -l $NETCDF | sed 's/  */ /g'`"
-    done 
+    done
     if [ "`echo $NETCDF | grep '..'`"  != "" ] ; then
       buff="`echo $NETCDF | sed -e 's/^\///' -e 's/\// /g'`"
       n=`echo $buff | wc -w`
@@ -294,7 +298,7 @@ if [ -n "$NETCDF4" ] ; then
       while [ $i -le $n ]
       do
         if [ "`echo $buff | cut -d' ' -f$i`" = ".." ] ; then
-          m=$i 
+          m=$i
         fi
         i=$(( $i + 1 ))
       done
@@ -353,12 +357,12 @@ else
       unset WRFPLUS_DIR
    fi
 fi
-# Users who are cross-compiling can set environment variable 
-# $WRF_OS to override the value normally obtained from `uname`.  
-# If $WRF_OS is set, then $WRF_MACH can also be set to override 
-# the value normally obtained from `uname -m`.  If $WRF_OS is 
-# set and $WRF_MACH is not set, then $WRF_MACH defaults to "ARCH".  
-# If $WRF_OS is not set then $WRF_MACH is ignored.  
+# Users who are cross-compiling can set environment variable
+# $WRF_OS to override the value normally obtained from `uname`.
+# If $WRF_OS is set, then $WRF_MACH can also be set to override
+# the value normally obtained from `uname -m`.  If $WRF_OS is
+# set and $WRF_MACH is not set, then $WRF_MACH defaults to "ARCH".
+# If $WRF_OS is not set then $WRF_MACH is ignored.
 if [ -n "$WRF_OS" ] ; then
   echo "${0}:  WRF operating system set to \"${WRF_OS}\" via environment variable \$WRF_OS"
   os=$WRF_OS
@@ -411,7 +415,7 @@ fi
 # the "!" is removed by Config.pl
 if [ -n "$WRF_HYDRO" ] ; then
   if [ $WRF_HYDRO = 1 ] ; then
-      echo building WRF-HYDRO 
+      echo building WRF-HYDRO
       compileflags="${compileflags}!-DWRF_HYDRO"
       echo $compileflags
   fi
@@ -487,11 +491,11 @@ if [ -n "$WRF_CHEM" ] ; then
     echo building WRF with chemistry option
     compileflags="${compileflags}!-DWRF_CHEM!-DBUILD_CHEM=1"
     if [ -n "$WRF_KPP" ] ; then
-      if [ $WRF_KPP = 1 ] ; then    
+      if [ $WRF_KPP = 1 ] ; then
         echo building WRF with KPP chemistry option
         compileflags="${compileflags}!-DWRF_KPP"
       fi
-    fi 
+    fi
   else
     compileflags="${compileflags} "
   fi
@@ -552,7 +556,7 @@ if test -n "$PERL" ; then
    srch=`grep -i "^#ARCH.*$os" arch/configure.defaults | grep -i "$mach"`
    if [ -n "$srch" ] ; then
      $PERL arch/Config.pl -dmparallel=$COMMLIB -ompparallel=$OMP -perl=$PERL \
-          -netcdf=$NETCDF -pnetcdf=$PNETCDF -hdf5=$HDF5 -phdf5=$PHDF5 -os=$os -mach=$mach -ldflags=$ldflags \
+          -netcdf=$NETCDF -netcdff=$NETCDFF -pnetcdf=$PNETCDF -hdf5=$HDF5 -phdf5=$PHDF5 -os=$os -mach=$mach -ldflags=$ldflags \
           -compileflags=$compileflags -opt_level=$opt_level -USENETCDFF=$USENETCDFF -USENETCDF=$USENETCDF \
           -time=$FORTRAN_COMPILER_TIMER -tfl="$TFL" -cfl="$CFL" -config_line="$config_line" \
           -wrf_core=$wrf_core -gpfs=$GPFS_PATH -curl=$CURL_PATH -dep_lib_path="$DEP_LIB_PATH"
@@ -600,7 +604,7 @@ if test -f configure.wrf ; then
   if [ "$os" = "Darwin" ] ; then
     # fseeko64 does not exist under Darwin fseeko does. Remove the 0 length executable
     # file that might get generated anyway, even though the compiler complains about missing reference.
-    /bin/rm -f tools/fseeko64_test 
+    /bin/rm -f tools/fseeko64_test
   fi
   if test -x tools/fseeko64_test ; then
     ( tools/fseeko64_test 2> /dev/null ) 1> /dev/null
@@ -634,18 +638,18 @@ echo "If you wish to change settings, please edit that file."
 echo "If you wish to change the default options, edit the file:"
 echo "     arch/configure.defaults"
 
-if test -n "$NETCDF" ; then
-  if [ ! -f $NETCDF/include/netcdf.inc ] ; then
+if test -n "$NETCDFF" ; then
+  if [ ! -f $NETCDFF/include/netcdf.inc ] ; then
     echo
-    echo "Error : Not found $NETCDF/include/netcdf.inc"
+    echo "Error : Not found $NETCDFF/include/netcdf.inc"
     echo "        Please check this installation of NetCDF and re-run this configure script"
     exit -1
   fi
-  grep nf_format_64bit $NETCDF/include/netcdf.inc > /dev/null
+  grep nf_format_64bit $NETCDFF/include/netcdf.inc > /dev/null
   configure_aaaa=$? ; export configure_aaaa
   if [ $configure_aaaa -a -z "$WRFIO_NCD_NO_LARGE_FILE_SUPPORT" ] ; then
     echo "NetCDF users note:"
-    echo " This installation of NetCDF supports large file support.  To DISABLE large file" 
+    echo " This installation of NetCDF supports large file support.  To DISABLE large file"
     echo " support in NetCDF, set the environment variable WRFIO_NCD_NO_LARGE_FILE_SUPPORT"
     echo " to 1 and run configure again. Set to any other value to avoid this message."
   fi
@@ -670,7 +674,7 @@ if [ -n "$WRFPLUS" ] ; then
   fi
 fi
 
-#Checking cross-compiling capability for some particular environment 
+#Checking cross-compiling capability for some particular environment
 #on Linux and Mac box
 
 if [ $os = "Linux" -o $os = "Darwin" ]; then
@@ -704,7 +708,7 @@ EOF
     rm ${foo} ${foo}.F 2> /dev/null
   fi
 
-cat > ${foo}.c <<EOF 
+cat > ${foo}.c <<EOF
  int main(int argc, char ** argv)
  {
      return (0);
@@ -752,7 +756,7 @@ EOF
     netcdf_arch="`file ${foo}.o | grep -o -E '[0-9]{2}-bit|i386'`"
     rm ${foo}.o
 
-    $SFC -o ${foo} ${foo}.f > /dev/null 2>&1 
+    $SFC -o ${foo} ${foo}.f > /dev/null 2>&1
     SFC_arch="`file ${foo} | grep -o -E '[0-9]{2}-bit|i386'`"
     rm ${foo} ${foo}.o 2> /dev/null
 
@@ -770,7 +774,7 @@ EOF
     if [ "$SFC_arch" = "" -o "$SCC_arch" = "" -o "$CCOMP_arch" = "" ]; then
       echo "  One of compilers testing failed!"
       echo "  Please check your compiler"
-      echo 
+      echo
       rm -f ${foo} ${foo}.[cfo] 2> /dev/null
       exit
     else
@@ -881,14 +885,14 @@ EOF
             echo Please check your NetCDF lib and compiler
         else
           echo -m64 is appended to configure.wrf
-          echo It will be forced to build in 64-bit. 
+          echo It will be forced to build in 64-bit.
           echo If you don\'t want 64-bit binaries, please use 32-bit NetCDF, and re-run the configure script.
         fi
       fi
       ;;
     esac
 
-    if [ -e configure.wrf.edit ]; then 
+    if [ -e configure.wrf.edit ]; then
       mv configure.wrf.edit configure.wrf
     fi
 
@@ -1046,7 +1050,7 @@ if [ -n "$NETCDF4" ] ; then
       echo "OR set NETCDF_classic variable                                               "
       echo "   bash/ksh : export NETCDF_classic=1
       echo "        csh : setenv NETCDF_classic 1
-      echo 
+      echo
       echo "Then re-run this configure script                                            "
       echo
       echo "!!! configure.wrf has been REMOVED !!!"
diff --git a/external/io_netcdf/makefile b/external/io_netcdf/makefile
index fa638d45..2bb17fb1 100644
--- a/external/io_netcdf/makefile
+++ b/external/io_netcdf/makefile
@@ -2,17 +2,17 @@
 
 OBJSL   = wrf_io.o field_routines.o module_wrfsi_static.o
 OBJS    = $(OBJSL)
-CODE    = ext_ncd_get_dom_ti.code ext_ncd_get_var_td.code ext_ncd_get_var_ti.code ext_ncd_put_dom_ti.code ext_ncd_put_var_td.code ext_ncd_put_var_ti.code transpose.code 
-FFLAGS  =  $(FCFLAGS) -I$(NETCDFPATH)/include -I../ioapi_share
-LIBS    = $(LIB_LOCAL) -L$(NETCDFPATH)/lib -lnetcdf
-LIBFFS  = $(LIB_LOCAL) -L$(NETCDFPATH)/lib -lnetcdff -lnetcdf $(NETCDF4_DEP_LIB)
+CODE    = ext_ncd_get_dom_ti.code ext_ncd_get_var_td.code ext_ncd_get_var_ti.code ext_ncd_put_dom_ti.code ext_ncd_put_var_td.code ext_ncd_put_var_ti.code transpose.code
+FFLAGS  =  $(FCFLAGS) -I$(NETCDFFPATH)/include -I../ioapi_share
+LIBS    = $(LIB_LOCAL) -L$(NETCDFPATH)/lib -L$(NETCDFPATH)/lib64 -lnetcdf
+LIBFFS  = $(LIB_LOCAL) -L$(NETCDFPATH)/lib -L$(NETCDFPATH)/lib64 -L$(NETCDFFPATH)/lib -L$(NETCDFFPATH)/lib64 -lnetcdff -lnetcdf $(NETCDF4_DEP_LIB)
 CPP1    = $(CPP) -P $(TRADFLAG)
 M4      = m4 -Uinclude -Uindex -Ulen
 AR      = ar
 
 .SUFFIXES:      .F90 .f .o .code
 
-all : libwrfio_nf.a 
+all : libwrfio_nf.a
 
 libwrfio_nf.a:         $(OBJS) $(CODE)
 			/bin/rm -f $@
@@ -24,7 +24,7 @@ libwrfio_nf.a:         $(OBJS) $(CODE)
 			$(RANLIB) $@
 
 wrf_io.o:               wrf_io.F90 $(CODE)
-			grep nf_format_64bit $(NETCDFPATH)/include/netcdf.inc ;\
+			grep nf_format_64bit $(NETCDFFPATH)/include/netcdf.inc ;\
                         a=$$? ; export a ; \
 			if [ $$a -a "$$WRFIO_NCD_LARGE_FILE_SUPPORT" = "1" ] ; then \
                           $(CPP1) -DWRFIO_NCD_LARGE_FILE_SUPPORT -I../ioapi_share wrf_io.F90 | $(M4) - > wrf_io.f ; \
@@ -43,14 +43,14 @@ diffwrf:                diffwrf.F90
 	x=`echo "$(FC)" | awk '{print $$1}'` ; export x ; \
 	if [ $$x = "gfortran" ] ; then \
            echo removing external declaration of iargc for gfortran ; \
-	   $(CPP1) -I$(NETCDFPATH)/include -I../ioapi_share diffwrf.F90 | sed '/integer *, *external.*iargc/d' > diffwrf.f ;\
+	   $(CPP1) -I$(NETCDFFPATH)/include -I../ioapi_share diffwrf.F90 | sed '/integer *, *external.*iargc/d' > diffwrf.f ;\
         else \
-	   $(CPP1) -I$(NETCDFPATH)/include -I../ioapi_share diffwrf.F90 > diffwrf.f ; \
+	   $(CPP1) -I$(NETCDFFPATH)/include -I../ioapi_share diffwrf.F90 > diffwrf.f ; \
         fi
 	$(FC) -c $(FFLAGS) diffwrf.f
 	@if [ \( -f ../../frame/wrf_debug.o \) -a \( -f ../../frame/module_wrf_error.o \) -a \( -f $(ESMF_MOD_DEPENDENCE) \) -a \( -f ../../frame/clog.o \) ] ; then \
 	  echo "diffwrf io_netcdf is being built now. " ; \
-          if [ \( -f $(NETCDFPATH)/lib/libnetcdff.a -o -f $(NETCDFPATH)/lib/libnetcdff.so -o -f $(NETCDFPATH)/lib/libnetcdff.dll.a \) ] ; then \
+          if [ \( -f $(NETCDFFPATH)/lib/libnetcdff.a -o -f $(NETCDFFPATH)/lib/libnetcdff.so -o -f $(NETCDFFPATH)/lib/libnetcdff.dll.a \) ] ; then \
             $(FC) $(FFLAGS) $(LDFLAGS) -o diffwrf diffwrf.o $(OBJSL) ../../frame/wrf_debug.o ../../frame/module_wrf_error.o ../../frame/clog.o $(ESMF_IO_LIB_EXT) $(LIBFFS) ;\
           else \
             $(FC) $(FFLAGS) $(LDFLAGS) -o diffwrf diffwrf.o $(OBJSL) ../../frame/wrf_debug.o ../../frame/module_wrf_error.o ../../frame/clog.o $(ESMF_IO_LIB_EXT) $(LIBS) ;\
diff --git a/external/io_pnetcdf/Makefile b/external/io_pnetcdf/Makefile
index f7d1d365..10b0ed86 100644
--- a/external/io_pnetcdf/Makefile
+++ b/external/io_pnetcdf/Makefile
@@ -8,16 +8,16 @@ CODE    = ext_pnc_get_dom_ti.code \
 	  ext_pnc_put_dom_ti.code \
 	  ext_pnc_put_var_td.code \
 	  ext_pnc_put_var_ti.code \
-	  transpose.code 
-FFLAGS  = $(FCFLAGS) -I$(NETCDFPATH)/include -I../ioapi_share
-LIBS    = -L$(NETCDFPATH)/lib -lpnetcdf
+	  transpose.code
+FFLAGS  = $(FCFLAGS) -I$(NETCDFFPATH)/include -I../ioapi_share
+LIBS    = -L$(NETCDFFPATH)/lib -lpnetcdf
 CPP1    = $(CPP) -P $(TRADFLAG)
 M4      = m4 -Uinclude -Uindex -Ulen
 AR      = ar
 
 .SUFFIXES:      .F90 .f .o .code
 
-all : libwrfio_pnf.a 
+all : libwrfio_pnf.a
 
 libwrfio_pnf.a:		$(OBJS) $(CODE)
 			/bin/rm -f libwrfio_pnf.a
@@ -25,15 +25,15 @@ libwrfio_pnf.a:		$(OBJS) $(CODE)
 			$(RANLIB) libwrfio_pnf.a
 
 wrf_io.o:               wrf_io.F90 $(CODE)
-			$(CPP1) -I$(NETCDFPATH)/include -I../ioapi_share wrf_io.F90 | $(M4) - > wrf_io.f
+			$(CPP1) -I$(NETCDFFPATH)/include -I../ioapi_share wrf_io.F90 | $(M4) - > wrf_io.f
 			$(FC) $(FFLAGS) -c wrf_io.f
 
 module_wrfsi_static.o:  module_wrfsi_static.F90
-			$(CPP1) -I$(NETCDFPATH)/include -I../ioapi_share module_wrfsi_static.F90 > module_wrfsi_static.f
+			$(CPP1) -I$(NETCDFFPATH)/include -I../ioapi_share module_wrfsi_static.F90 > module_wrfsi_static.f
 			$(FC) $(FFLAGS) -c module_wrfsi_static.f
 
 field_routines.o:	field_routines.F90 wrf_io.o
-			$(CPP1) -I$(NETCDFPATH)/include -I../ioapi_share field_routines.F90 > field_routines.f
+			$(CPP1) -I$(NETCDFFPATH)/include -I../ioapi_share field_routines.F90 > field_routines.f
 			$(FC) $(FFLAGS) -c field_routines.f
 
 superclean:
diff --git a/main/depend.common b/main/depend.common
index 4b8f2f9a..6839c10d 100644
--- a/main/depend.common
+++ b/main/depend.common
@@ -722,6 +722,7 @@ module_radiation_driver.o: \
                 module_ra_flg.o \
                 module_ra_aerosol.o \
                 module_mp_thompson.o \
+		module_ra_eclipse.o \
 		../frame/module_driver_constants.o \
 		../frame/module_state_description.o \
 		../frame/module_dm.o \
